src/main/java/org/jenkinsci/plugins/gitclient/CliGitAPIImpl.java

```diff
@@ -1186,12 +1191,9 @@ else if (!referencePath.isDirectory())
                 Pattern pattern = Pattern.compile(SUBMODULE_REMOTE_PATTERN_STRING, Pattern.MULTILINE);
                 Matcher matcher = pattern.matcher(cfgOutput);
 
-                ExecutorService executorService;
-                if (threads > 1) {
-                    executorService = Executors.newFixedThreadPool(threads);
-                } else {
-                    executorService = Executors.newSingleThreadExecutor();
-                }
+                List<Callable<String>> launchCommands = new ArrayList<>();
+                ExceptionCatchingThreadFactory threadFactory = new ExceptionCatchingThreadFactory(new NamingThreadFactory(new DaemonThreadFactory(), "GitSubmoduleUpdate"));
+                ExecutorService executorService = threads > 1 ? Executors.newFixedThreadPool(threads, threadFactory) : Executors.newSingleThreadExecutor();
```

Comment:
I was still thinking more about the default case (threads = 1). I would really like to behave it like before - without any new threads beeing spawned, to reduce non-required resource costs. There may already be a simple solution to this - replace Java's `Executors.newSingleThreadExecutor()` with Guava's `MoreExecutors.sameThreadExecutor()`.

Reply:
Adding guava as a git plugin dependency seems like a high price to pay for the case of a single submodule.  Have I missed that it is already an existing dependency somehow?

Reply:
Guava (package `com.google.common`) is already used in some classes.
Its provided via the parent pom: `org.jenkins-ci.plugins:plugin:pom:3.19` -> `org.jenkins-ci.main:jenkins-core:jar:2.60.3` ->`com.google.guava:guava:jar:11.0.1`

Reply:
Sounds good, I also think it makes sense to continue in the current thread in this case. I will remove the ternary one-liner refactoring as well.